js/client/src/components/app/Graph.js

```diff
@@ -14,20 +14,42 @@ const Graph = ({ polygonData, darkMode }) => {
   const [title, setTitle] = useState({ value: 'make', label: 'Make of Vehicle' });
 
   const fetchGraph = async () => {
-    const response = await axios
-      .get(`${API_URL}/api/citation/graph`, {
-        params: {
-          polygon: polygonData,
-          filterBy: selectedKey,
-        },
+    try {
+      if (Array.isArray(polygonData)) {
+        const response = await axios
+          .get(`${API_URL}/api/citation/graph`, {
+            params: {
+              polygon: polygonData,
+              filterBy: selectedKey,
+            },
+          })
+          
+        var parsed = response.data.map((obj) => {
+          obj["y"] = (parseInt(obj.y))
+          return obj
+        })
+
+      } else {
+        const response = await axios
+        .get(`${API_URL}/api/citation/graph/zip`, {
+          params: {
+            zip: polygonData,
+            filterBy: selectedKey,
+          },
+        })
+        
+      var parsed = response.data.map((obj) => {
+        obj["y"] = (parseInt(obj.y))
+        return obj
       })
-      
-    var parsed = response.data.map((obj) => {
-      obj["y"] = (parseInt(obj.y))
-      return obj
-    })
 
-    setData(parsed)
+      }
+  
+      setData(parsed)
+
+    } catch (err){
+      console.log(err)
```

Comment:
I'm thinking it maybe good to create a unified error handler with toast message across the whole app.